src/Features/Core/SolutionCrawler/WorkCoordinator.LowPriorityProcessor.cs

```diff
@@ -76,6 +73,39 @@ protected override Task WaitAsync(CancellationToken cancellationToken)
                         }
                     }
 
+                    private async Task WaitForHigherPriorityOperationsAsync()
+                    {
+                        do
+                        {
+                            // we wait for global operation if there is anything going on
+                            var globalOperationTask = GlobalOperationWaitAsync();
+
+                            // we wait for higher processor to finish if it is running
+                            var highPriorityTask = this.Processor._highPriorityProcessor.Running;
+
+                            // we wait for normal processor to finish if it is running
+                            var normalPriorityTask = this.Processor._normalPriorityProcessor.Running;
+
+                            if (!globalOperationTask.IsCompleted ||
+                                !highPriorityTask.IsCompleted ||
+                                !normalPriorityTask.IsCompleted)
+                            {
+                                await Task.WhenAll(globalOperationTask, highPriorityTask, normalPriorityTask).ConfigureAwait(false);
+                            }
+
+                            // if there are no more work left for higher queue, then it is our time to go ahead
+                            if (!HigherQueueHasWorkItem)
```

Comment:
Do we really need a while loop to do idle processing here, we already have idle processing loop in base type right?

Another approach would be to change this method to return `Task<bool>` and return value `HigherQueueHasWorkItem`. Based on this value, the caller can go ahead with execution if HigherQueueHasWorkItem = false or return early with no-op otherwise?